### PR TITLE
Added Azure support for safety controller

### PR DESCRIPTION
**Pull request by prashanth26**
This commit adds azure support in safety controller.

### CODE REVIEW

File: pkg/apis/machine/validation/azuremachineclass.go [hunk 1]
```diff
@@ -60,6 +62,30 @@ func validateAzureMachineClassSpec(spec *machine.AzureMachineClassSpec, fldPath
...
+	return allErrs
+}
+
+func validateAzureClassSpecTags(tags map[string]string, fldPath *field.Path) field.ErrorList {
+	allErrs := field.ErrorList{}
+	clusterName := ""
+	nodeRole := ""
+
+	for key := range tags {
+		if strings.Contains(key, "kubernetes.io-cluster-") {
```
Comment (hardikdr) [inline pkg/apis/machine/validation/azuremachineclass.go]: Update doku as well.

File: pkg/driver/driver_azure.go [hunk 2]
```diff
@@ -193,8 +201,64 @@ func (d *AzureDriver) GetExisting() (string, error) {
...
+	if searchClusterName == "" ||
+		searchNodeRole == "" ||
+		d.AzureMachineClass.Spec.ResourceGroup == "" {
+		return listOfVMs
+	}
+
+	d.setup()
+	result, err := vmClient.List(d.AzureMachineClass.Spec.ResourceGroup)
+	if err != nil {
+		glog.Errorf("Failed to list VMs. Error Message - %s", err)
```
Comment (hardikdr) [inline pkg/driver/driver_azure.go]: return here ?

File: pkg/apis/machine/validation/azuremachineclass.go (hunk 1 above)
Comment (prashanth26) [reply pkg/apis/machine/validation/azuremachineclass.go]: Done.

File: pkg/driver/driver_azure.go (hunk 2 above)
Comment (prashanth26) [reply pkg/driver/driver_azure.go]: Done.

**Pull request merged**